src/zope/server/serverbase.py

```diff
@@ -42,15 +42,19 @@ def __init__(self, ip, port, task_dispatcher=None, adj=None, start=1,
         self.port = port
         self.task_dispatcher = task_dispatcher
         self.create_socket(socket.AF_INET, socket.SOCK_STREAM)
-        self.set_reuse_addr()
-        self.bind((ip, port))
-        self.verbose = verbose
-        self.hit_log = hit_log
-        self.logger = logging.getLogger(self.__class__.__name__)
-        self.server_name = self.computeServerName(ip)
-
-        if start:
-            self.accept_connections()
+        try:
+            self.set_reuse_addr()
+            self.bind((ip, port))
+            self.verbose = verbose
+            self.hit_log = hit_log
+            self.logger = logging.getLogger(self.__class__.__name__)
+            self.server_name = self.computeServerName(ip)
```

Comment:
I would prefer that assignments not related to the exception being raised be done outside the `try:` (probably before the call to `self.create_socket`).

Reply:
I was thinking that `computeServerName()` might raise something unexpected.

I'm move the other assignments up.